### PR TITLE
implement initializeDots method

### PR DESCRIPTION
**Pull request by afazio1**

### CODE REVIEW

**Review by Maiikiru — CHANGES_REQUESTED** [review]
Amazing job! However, before I merge, I just saw a couple of things I saw that you may (or may not) have considered.
1) dotCount is actually the number of dots the player has collected from what im seeing (i know it's bad naming, but dont blame me for it) . Therefore I dont think you should increment it when initializing. (please correct me if im wrong)
2) Rand is already defined as an attribute of the class, is there a particular reason you redefined it?
3) I was thinking of moving the 35 on line 92 to a constant so we can reduce "magic numbers" I dont know if you agree with this decision.

Overall, besides those simple things, everything else looks good to me.

File: app/src/main/java/com/example/demo_2340/GameActivity.java [hunk 1]
```diff
@@ -83,7 +83,15 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
...
     private void initializeDots() {
-        // TODO Create and add dots with random positions
+        Random rand = new Random();
+        int maxX = screenWidth;
+        int maxY = screenHeight;
+        for (int i = 0; i < MAX_DOTS; i++) {
+            int x = rand.nextInt(maxX);
+            int y = rand.nextInt(maxY);
+            dots.add(new Dot(x, y, 35));
+            dotCount++;
```
Comment (Maiikiru) [inline app/src/main/java/com/example/demo_2340/GameActivity.java:93]: I think this is unnecessary as dotCount is the number of dots that the player has collected (please correct me if im wrong)

Comment (Maiikiru) [inline app/src/main/java/com/example/demo_2340/GameActivity.java:93]: I think you should remove this line (explanation in the review I left)

File: app/src/main/java/com/example/demo_2340/GameActivity.java [hunk 2]
```diff
@@ -83,7 +83,15 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
     }
 
     private void initializeDots() {
-        // TODO Create and add dots with random positions
+        Random rand = new Random();
```
Comment (Maiikiru) [inline app/src/main/java/com/example/demo_2340/GameActivity.java:86]: I think you dont need rand, but can use random as defined above, but if you want to leave it, it's no problem

File: app/src/main/java/com/example/demo_2340/GameActivity.java [hunk 3]
```diff
@@ -83,7 +83,15 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
...
+        int maxX = screenWidth;
+        int maxY = screenHeight;
+        for (int i = 0; i < MAX_DOTS; i++) {
+            int x = rand.nextInt(maxX);
+            int y = rand.nextInt(maxY);
+            dots.add(new Dot(x, y, 35));
```
Comment (Maiikiru) [inline app/src/main/java/com/example/demo_2340/GameActivity.java:92]: I want to eliminate the 35 "magic number" if possible.

File: app/src/main/java/com/example/demo_2340/GameActivity.java (hunk 1 above)
Comment (afazio1) [reply app/src/main/java/com/example/demo_2340/GameActivity.java:93]: I think this method is incorrect as well if `dotCount` represents the total dots player-collected dots. 
```java
// Maintains 20 dots on screen
    private void respawnDotsIfNeeded() {
        // TODO: if dots drop below 20, respawn dots
        while(dotCount < 20){
            respawnDot();
        }
    }
```

Comment (Maiikiru) [reply app/src/main/java/com/example/demo_2340/GameActivity.java]: yeah, I will fix that, I think i messed that one up

**Review by Maiikiru — APPROVED** [review]
Great work

**Pull request merged**